paramiko/ssh_exception.py

```diff
@@ -233,3 +233,13 @@ class ConfigParseError(SSHException):
     """
 
     pass
+
+
+class AuthenticationTimeoutException(SSHException):
+    """
+    Exception raised when an authentication timeout occurs.
+
+    .. versionadded:: 3.1.x
```

Comment:
I left the version string as `3.1.x` because I didn't want to assume the next version is `3.1.1`.